apps/api/src/app/workflows-v2/dto/workflow-commons-fields.ts

```diff
@@ -0,0 +1,79 @@
+import { IsArray, IsBoolean, IsDefined, IsObject, IsOptional, IsString } from 'class-validator';
+
+import { JsonSchema } from '@novu/framework';
+import { StepTypeEnum, WorkflowPreferences } from '@novu/shared';
+import { WorkflowResponseDto } from './workflow-response-dto';
+
+export class ControlsSchema {
+  schema: JsonSchema;
+}
+
+export type StepResponseDto = StepDto & {
+  stepUuid: string;
+};
+
+export type StepUpdateDto = StepDto & {
+  stepUuid: string;
+};
+
+export type StepCreateDto = StepDto;
+
+export type ListWorkflowResponse = {
+  workflowSummaries: MinifiedResponseWorkflowDto[];
+  totalResults: number;
+};
+
+export type MinifiedResponseWorkflowDto = Pick<
```

Comment:
Please make the naming very specific to its use case.

```suggestion
export type WorkflowListResponseDto = Pick<
```